### DISCUSSION TITLE
Unable to toggle switch on UI MSGR: Command 'p1b2/val' not found => 1

### DISCUSSION THREAD

**Discussion by Tall-Paul** [HASwitchPlate/openHASP#675] [General]
Hi all, 

Just started playing around with OpenHasp and have hit this problem.  I'm running 0.7.0-rc11 on an Guition ESP32-S3-4848S040 and using the homeassistant plugin v0.7.2.

I've sorted out the UI, uploading a jsonl file, I can toggle an entity in HA by toggling the button on the UI...all great!

However, I can't get the UI to change in response to a HomeAssistant event.

This is my jsonl file:

{"page": 1,"id": 2,"obj": "btn","x": 200,"y": 40,"w": 105,"h": 90,"toggle": true,"text": "","text_font": 32,"align": 1}

And, to remove the homeassistant part of the problem, I'm using MQTT explorer to just send a payload of 1 to hasp/plate1/command/p1b2.val

In the log I see this:

Prompt > [22:26:48.222][94196/111632 15][57732/57940  1] MQTT RCV: p1b2/val = 1
Prompt > [22:26:48.232][94196/111632 15][57732/57940  1] MSGR: Command 'p1b2/val' not found => 1

**Comment by darkman7hr** [comment]
'p1b2/val'  = 'p1b2.val'  ?

**Reply by Tall-Paul** [reply]
Yup, I send p1b2.val and it shows p1b2/val in the log.

**Comment by Tall-Paul** [comment]
Okay, I think I know what the problem is; I'm using RabbitMQ as an MQTT broker.  RabbitMQ does not like '.' as a topic seperator and (it seems) might replace them with a slash.

Right, off to install mosquitto instead.

**Comment by Tall-Paul** [comment]
Confirmed: Same setup with Mosquitto as the broker works perfectly.

Might want to mention somewhere (unless the docs already do and I missed it, entirely possible!) that openHasp does not support using rabbitmq as the broker.

**Comment by fvanroie** [comment]
There are workarounds that don't use the dot notation in the topic, like:
`hasp/plate1/command` => `p1b2.val=1` payload
`hasp/plate1/command/jsonl` => `{"page":1,"id":2,"val":1}` payload
